File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.java
class Solution {
    public int strStr(String haystack, String needle) {
        int hlen=haystack.length();
        int nlen=needle.length();

        for (int i =0; i<=hlen-nlen; i++){
            int j=0;

            while (j< nlen && haystack.charAt(i+j)== needle.charAt(j)){
                j++;
            }
            if(j==nlen){
                return i;
            }
        }
        return -1;
    }

}

File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.js
/**
 * @param {string} haystack
 * @param {string} needle
 * @return {number}
 */
var strStr = function(haystack, needle) {
let T = haystack.length; 
let P = needle.length;

if(P===0){
    return 0;
}
 

 for(let i =0; i <=T-P; i++){
    let j =0;

    while (j < P && haystack.charAt(i+j)== needle.charAt(j)){
        j++;
    }

    if(j===P){
        return i;
    }
   
 }
  return -1
    
};